ros/src/computing/planning/motion/packages/astar_planner/nodes/velocity_set/velocity_set.cpp

```diff
@@ -65,42 +65,37 @@ void obstacleColorByKind(const EControl kind, std_msgs::ColorRGBA &color, const
 }
 
 // Display a detected obstacle
-void displayObstacle(const EControl& kind, const ObstaclePoints& obstacle_points, const ros::Publisher& obstacle_pub)
+void displayObstacles(EControl& kind, const std::vector<ObstacleInfo>& obstacle_infos, const ros::Publisher& obstacles_pub)
 {
+  visualization_msgs::MarkerArray marker_array;
   visualization_msgs::Marker marker;
   marker.header.frame_id = "/map";
   marker.header.stamp = ros::Time();
-  marker.ns = "my_namespace";
+  marker.ns = "obstacle";
   marker.id = 0;
   marker.type = visualization_msgs::Marker::CUBE;
   marker.action = visualization_msgs::Marker::ADD;
 
-  static geometry_msgs::Point prev_obstacle_point;
-  if (kind == EControl::STOP || kind == EControl::STOPLINE || kind == EControl::DECELERATE)
-  {
-    marker.pose.position = obstacle_points.getObstaclePoint(kind);
-    prev_obstacle_point = marker.pose.position;
-  }
-  else  // kind == OTHERS
+  for (auto& info : obstacle_infos)
   {
-    marker.pose.position = prev_obstacle_point;
+    marker.pose.position = info.points.getObstaclePoint(kind);
+    marker.scale.x = 1.0;
+    marker.scale.y = 1.0;
+    marker.scale.z = 2.0;
+    marker.lifetime = ros::Duration(0.1);
+    marker.frame_locked = true;
+    obstacleColorByKind(info.type, marker.color, 0.7);
+    marker_array.markers.push_back(marker);
   }
-  geometry_msgs::Quaternion quat;
-  marker.pose.orientation = quat;
```

Comment:
you need to add orientation otherwise rviz crashes because the orientation is not normalized

Reply:
but you have this error message, `Marker 'Stop Line/1' contains unnormalized quaternions` , so we need to focus here.
https://github.com/CPFL/Autoware/blob/741beb9b1a067c7593d507b2074abf39df9d9dbf/ros/src/computing/planning/motion/packages/astar_planner/nodes/velocity_set/velocity_set.cpp#L151